dashboard/src/components/QueryResult.tsx

```diff
@@ -17,25 +17,31 @@ export default function QueryResult() {
     </EmptyState>
   ) : (
     <VBox id='query-builder-result' customStyles='overflow-x-auto'>
-      {!!queryRes && typeof queryRes === 'string' && <Label>{queryRes}</Label>}
-      {!!queryRes && queryRes instanceof Array && !queryRes[0]?.result && (
-        <VBox>{<Table data={queryRes} />}</VBox>
-      )}
-      {!!queryRes &&
-        queryRes instanceof Array &&
-        queryRes[0]?.result instanceof Array &&
-        queryRes.filter(r => (r.result ?? []).length > 0).length > 0 && (
+      {!!editorTabs[editorSelectedTab].queryRes &&
```

Comment:
Instead of selecting out `editorTabs[editorSelectedTab]` every single time (not only is it harder to read, it is actually slower in JS because it can't optimize away the property selection when it's done by a variable like this). I'd recommend just above the `return` above to define something like:

```ts
const t = editorTabs[editorSelectedTab];
```

and then this becomes `{!!t.queryRest && typeof t.queryRes === 'string' ...`